### ISSUE TITLE
PermissionError: [Errno 13] Permission denied when packaging

### ISSUE THREAD

**Issue by alexgraeff** [opengisch/qfieldsync#382]
When trying to package a project for QField, I get a python error:

2022-02-02T20:10:46     WARNING    Traceback (most recent call last):
              File "C:\Users/admin/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\qfieldsync\gui\package_dialog.py", line 123, in package_project
              offline_convertor.convert()
              File "C:\Users/admin/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\qfieldsync\core\offline_converter.py", line 188, in convert
              copied_files = layer_source.copy(self.export_folder, copied_files)
              File "C:\Users/admin/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\qfieldsync\core\layer.py", line 234, in copy
              shutil.copy(os.path.join(source_path, basename + ext), dest_file)
              File "C:\PROGRA~1\QGIS32~2.3\apps\Python39\lib\shutil.py", line 418, in copy
              copyfile(src, dst, follow_symlinks=follow_symlinks)
              File "C:\PROGRA~1\QGIS32~2.3\apps\Python39\lib\shutil.py", line 282, in copyfile
              _copyfileobj_readinto(fsrc, fdst, min(file_size, COPY_BUFSIZE))
              File "C:\PROGRA~1\QGIS32~2.3\apps\Python39\lib\shutil.py", line 195, in _copyfileobj_readinto
              fdst_write(mv)
             PermissionError: [Errno 13] Permission denied
             

I was able to package a project once - got super excited about how it looked on my mobile, made some edits, tried to package a new project, and that's when this error started. 

I've tried starting with a fresh project, uninstalling and reinstalling QGIS, and I've tried 2 different machines. 

It initially worked on QGIS version 3.20.3 and subsequently failed on that version. An upgrade to 3.22.3 made no difference

**Comment by suricactus** [comment]
It seems you cannot use this part of the filesystem, thereby you get "PermissionError".

Can you select another destination for your packaged project? The safest option is to use your own "My Documents".

**Comment by alexgraeff** [comment]
I tried saving to my own "My Documents" folder and got the same error. I had also tried making sure to run as administrator.

**Comment by BentHaase** [comment]
Same issue for us, we are trying to export some files and get this error to any path, regardless of being local or remote (SMB / Dropbox).

Versions and Systems in use:


**QGIS version** `3.16.16-Hannover`
**QField Sync version** `3.4.4`
**Python version** `3.8.10 `
**OS version** `Ubuntu 20.04.4 LTS`

```python
2022-02-21T18:30:38     WARNING    Traceback (most recent call last):
              File "/home/<user>/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qfieldsync/gui/package_dialog.py", line 123, in package_project
              offline_convertor.convert()
              File "/home/<user>/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qfieldsync/core/offline_converter.py", line 188, in convert
              copied_files = layer_source.copy(self.export_folder, copied_files)
              File "/home/<user>/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qfieldsync/core/layer.py", line 234, in copy
              shutil.copy(os.path.join(source_path, basename + ext), dest_file)
              File "/usr/lib/python3.8/shutil.py", line 418, in copy
              copyfile(src, dst, follow_symlinks=follow_symlinks)
              File "/usr/lib/python3.8/shutil.py", line 264, in copyfile
              with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
              PermissionError: [Errno 13] Permission denied: '/home/<user>/Dropbox/some/QGIS/QGIS_Test/path/Export_QGIS/somename.xlsx'
```

**Comment by BentHaase** [comment]
> Same issue for us, we are trying to export some files and get this error to any path, regardless of being local or remote (SMB / Dropbox).
> 
> Versions and Systems in use:
> 
> **QGIS version** `3.16.16-Hannover` **QField Sync version** `3.4.4` **Python version** `3.8.10 ` **OS version** `Ubuntu 20.04.4 LTS`
> 
> ```python
> 2022-02-21T18:30:38     WARNING    Traceback (most recent call last):
>               File "/home/<user>/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qfieldsync/gui/package_dialog.py", line 123, in package_project
>               offline_convertor.convert()
>               File "/home/<user>/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qfieldsync/core/offline_converter.py", line 188, in convert
>               copied_files = layer_source.copy(self.export_folder, copied_files)
>               File "/home/<user>/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qfieldsync/core/layer.py", line 234, in copy
>               shutil.copy(os.path.join(source_path, basename + ext), dest_file)
>               File "/usr/lib/python3.8/shutil.py", line 418, in copy
>               copyfile(src, dst, follow_symlinks=follow_symlinks)
>               File "/usr/lib/python3.8/shutil.py", line 264, in copyfile
>               with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
>               PermissionError: [Errno 13] Permission denied: '/home/<user>/Dropbox/some/QGIS/QGIS_Test/path/Export_QGIS/somename.xlsx'
> ```

Okay fail on our side, one of the .xlsx tables used in the project were imported via mtp ("drag&drop") from the mobile device itself. This resulted in issues getting said file for an export as _mtp is not the most robust protocol_ on neither operating system. Using only local files in the project layers resolved the issue.

@alexgraeff You may also check if any of your project layers are stored on remote paths / remote devices which might not always be accessible. Accessing these files fails silently but the error above shows issue when writing said file to the export path.

**Comment by suricactus** [comment]
@BentHaase thanks for posting your solution. That was what I was trying to hint with "My Documents" part above.

**Comment by alexgraeff** [comment]
I don't know exactly what the issue was for me, but by trial and error I discovered that either a layer within one of my GeoPackages or the GeoPackage itself was causing issues. It looks to me like qfieldsync will package the entire GeoPackage, even if only one feature within it is in the project. Once I created a new GeoPackage with freshly generated feature layers, the problem went away. I had a few different GeoPackages loaded with this particular project and only one seemed to be causing the issue.

Unfortunately, the Python error given didn't hint at that at all, it took me several hours to figure it out.
